### DISCUSSION TITLE
Kafka failed to start after upgrade from 1.7.1 to 1.8.2

### DISCUSSION THREAD

**Discussion by Saruspete** [akvorado/akvorado#654] [Q&A]
Hello,

I tried to update akvorado from 1.7.1 to 1.8.2. i've merged/split the old configuration into the new split format in config/
Upon restart, kafka failed with error:
```
kafak java.lang.IllegalArgumentException: Error creating broker listeners from 'PLAINTEXT://:9092,CONTROLLER://:9093': No security protocol defined for listener CONTROLLER
```

So i tried to override it from env (inspired by .github/workflows/ci.yml):
```
KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CLIENT:PLAINTEXT,CONTROLLER:PLAINTEXT,EXTERNAL:PLAINTEXT
```
and the message was then the same, but for `PLAINTEXT` this time.
had to add `,PLAINTEXT:PLAINTEXT` to the configuration.
```
KAFKA_CFG_LISTENER_SECURITY_PROTOCOL_MAP=CLIENT:PLAINTEXT,CONTROLLER:PLAINTEXT,EXTERNAL:PLAINTEXT,PLAINTEXT:PLAINTEXT
```

 Maybe we should add the default value of the var too ? `PLAINTEXT:PLAINTEXT,SSL:SSL,SASL_PLAINTEXT:SASL_PLAINTEXT,SASL_SSL:SASL_SSL`

**Comment by vincentbernat** [comment]
This is because of https://github.com/bitnami/containers/issues/31488. Did you have an issue with duplicate data after all this? I would expect last data to be duplicated because of that (migration from Zookeeper where the data was to KRaft). I would have hoped this issue would have been fixed faster on Bitnami side for people to not run on it.

**Comment by Saruspete** [comment]
I don't see any duplicated data. Before finding the fix I've reverted from 3.3 to 2.8 to ensure all was still ok, then I've updated the var and jumped back to 3.3. I don't have duplicated data or missing exporter (checked with history value & librenms graphs).

As I've used the upstream version of docker-compose, I've also had the `KAFKA_ENABLE_KRAFT=no` added before migration.
During my tests, I tried to remove it to check, and it complained being in "version 0" while expecting version 1 (seems to come from `/var/lib/docker/volumes/akvorado_akvorado-kafka/_data/data/meta.properties`). 
So I added it again before fixing the listener.security.protocol.map variable.

**Comment by vincentbernat** [comment]
It seems that you have another bug in that when KRaft is disabled. You can fix it with `KAFKA_CFG_LISTENERS` set to `PLAINTEXT://:9092`.

**Comment by Saruspete** [comment]
Yes, so removing the CONTROLLER listener would be another possibility, but I couldn't find it anywhere in the docker-compose. 
Was it pulled from zookeeper, or was it built-in by default in the new images ?

**Reply by vincentbernat** [reply]
It was builtin by default in the new images.
